api/host.go

```diff
@@ -105,7 +105,12 @@ func (srv *Server) hostHandlerPOST(w http.ResponseWriter, req *http.Request, _ h
 			}
 		}
 	}
-	srv.host.SetInternalSettings(settings)
+	err := srv.host.SetInternalSettings(settings)
+	if err != nil {
+		// TODO: error returned is not necessarily because of a bad request.
+		writeError(w, err.Error(), http.StatusBadRequest)
```

Comment:
this is a common problem in our API, though the distinction is not very important. I don't think anyone checks the exact 400 code; only whether it's 200 or not.